src/zarr/core/group.py

```diff
@@ -1752,6 +1752,12 @@ def open(
         obj = sync(AsyncGroup.open(store, zarr_format=zarr_format))
         return cls(obj)
 
+    def __enter__(self) -> Self:
+        return self
+
+    def __exit__(self) -> None:  # noqa: PYI036
```

Comment:
```suggestion
    def __exit__(
        self,
        typ: type[BaseException] | None,
        exc: BaseException | None,
        tb: TracebackType | None
    ) -> None:
```

See https://docs.astral.sh/ruff/rules/bad-exit-annotation/

Reply:
updated in [d8f7d68](https://github.com/zarr-developers/zarr-python/pull/2691/commits/d8f7d686a15e3a62e87fb29f8d2288a08f4118ad). Also added tests.